### ISSUE TITLE
Use `bootc install to-filesystem` to build our bootimages

### ISSUE THREAD

**Issue by jlebon** [coreos/fedora-coreos-tracker#1827]
Notably, this moves a good chunk of the osbuild pipeline to bootc, including the partition table definition, formatting, and deploying of the commit.

Requires having bootc in all the streams first.

**Comment by cgwalters** [comment]
> Requires having bootc in all the streams first.

This is going to happen unconditionally soon as part of https://github.com/coreos/rpm-ostree/issues/4994 I think, at least if we just push that to f41

**Comment by cgwalters** [comment]
Something I'd note here is that for bootc I am trying really, really hard to "centralize" all the knowledge of how to deploy a bootc system in bootc itself, with the rationale that it's the thing performing "day 2" updates as well and we want to make sure that as a general rule "install" matches what happens on upgrades.

And we today have quite a lot of skew because *in parallel* the coreos team worked on using osbuild which has a bunch of fine-grained split up stages of which ostree is one step, while the bootc team worked on centralizing as much as possible under `bootc install to-filesystem` which is very opinionated by default.

This is also strongly related to coreos using bootc-image-builder as a container.

**Comment by cgwalters** [comment]
A really notable pain point that the general bootc world has that coreos doesn't is that for bootc we support both Anaconda and bootc-image-builder which share...very little code except underlying fundamental utilities like `mkfs` etc. So centralizing the logic in `bootc install to-filesystem` helps force more code sharing between those two wildly different things.

**Comment by cverna** [comment]
> And we today have quite a lot of skew because _in parallel_ the coreos team worked on using osbuild which has a bunch of fine-grained split up stages of which ostree is one step, while the bootc team worked on centralizing as much as possible under `bootc install to-filesystem` which is very opinionated by default.

Yes the work on osbuild has been driven by business requests to align on Image Builder within Red Hat, and at that time bootc-image-builder was just starting. The user need of allowing users to generate their own  disk images of CoreOS is still here, are there currently any blockers for using bootc-image-builder for CoreOS?

**Comment by cverna** [comment]
Not directly related but I just stumbled on https://github.com/ublue-os/titanoboa

**Comment by jlebon** [comment]
Let's track the bootc-image-builder part of this discussion in this higher-level issue: https://github.com/coreos/fedora-coreos-tracker/issues/1906

**Comment by dustymabe** [comment]
PoC in https://github.com/coreos/coreos-assembler/pull/4224

**Comment by cgwalters** [comment]
> I think the main thing to look into is the aleph file content

I wouldn't get too hung up on this one, it's probably simplest to just have both for the near term future.

**Comment by jlebon** [comment]
I think we should keep the sentinel UUIDs for boot and root and continue relying on restamping on first boot (and not have baked `root` and `boot` kargs).

**Comment by travier** [comment]
https://github.com/coreos/coreos-assembler/pull/4224

**Comment by travier** [comment]
This is now enabled for the `rawhide` stream with:
- https://github.com/coreos/coreos-assembler/pull/4224
- https://github.com/coreos/fedora-coreos-config/pull/4093
- https://github.com/coreos/fedora-coreos-config/pull/4118

**Comment by jbtrystram** [comment]
Enabling in  the `next` stream : https://github.com/coreos/fedora-coreos-config/pull/4272
